### PR TITLE
fix(button): add higher contrast danger button color token to all themes

### LINKED ISSUE #3736: [Button] Support token & danger buttons fix
## Three tasks to make Danger button accessible again! 

- [x] Update `$support-01` token colors on dark themes. Final token values should be below. **This token is for error message text, 4.5:1 accessible.** 

| White | G10 | G90 | G100 |
| --- | --- | --- | --- |
| Red 60 | Red 60 | Red 50 | Red 50 |  

- [x] Add `$danger` token with below values. **This token is for danger button / icon / non-text usage, 3:1 accessible.** 

| White | G10 | G90 | G100 |
| --- | --- | --- | --- |
| Red 60 | Red 60 | Red 60 | Red 60 |  

- [ ] Fix Danger button's color to use `$danger` or Red 60 on all themes. It's currently using an out-of-palette color for accessibility sake, but now our palette's Red 60 should be. 

image

### PR DESCRIPTION
**Pull request by dakahn**
Closes #3736 

Some updates using the new danger color token 🔴 for a11y.
#### Changelog

- add the danger variable to White, g10, g90, g100
- add the token to our token list
- replace hard coded out of palette color with new danger token in button

Tests are broken here, working on fixing them up 👍 

#### Instruction for reviewers:

- Confirm danger button component is using $danger token for background.
- All tests pass, no other components are affected.

### CODE REVIEW

**Comment by dakahn** [comment]
@joshblack looking at the failing tests. Are we adding these to v9 as well? Also what's the format for updating metadata.yml -- I'm assuming that's what the referenced "meta" is in the other failing test.

**Review by emyarod — COMMENTED** [review]
it looks like `meta` is referring to the `unstable__meta` object in `tokens.js`. it's currently missing the 'danger' token

**Comment by joshblack** [comment]
@dakahn right, make sure the token is included in all themes and that the name is included in the `colors` array in `tokens` 👍

**Comment by jnm2377** [comment]
Also can we update the commit title before merging? 💯 Thanks!

**Review by joshblack — APPROVED** [review]

File: packages/components/src/globals/scss/vendor/@carbon/themes/scss/_tokens.scss [hunk 1]
```diff
@@ -313,3 +313,7 @@ $active-01: map-get($carbon--theme, 'active-01') !default;
 /// @alias hover-ui
 /// @deprecated
 $hover-field: map-get($carbon--theme, 'hover-field') !default;
+/// @type Color
+/// @access public
+/// #group @carbon/themes
```
Comment (joshblack) [inline packages/components/src/globals/scss/vendor/@carbon/themes/scss/_tokens.scss]:
```suggestion
/// @group @carbon/themes
```

Does it need to be like this?

File: packages/components/src/globals/scss/vendor/@carbon/themes/scss/_tokens.scss [hunk 2]
```diff
@@ -321,3 +321,7 @@ $active-01: map-get($carbon--theme, 'active-01') !default;
 /// @alias hover-ui
 /// @deprecated
 $hover-field: map-get($carbon--theme, 'hover-field') !default;
+/// @type Color
+/// @access public
+/// @group @carbon/themes
+$danger: map-get($carbon--theme, 'danger') !default;
```
Comment (joshblack) [inline packages/components/src/globals/scss/vendor/@carbon/themes/scss/_tokens.scss]: Ah wait, this should be included automatically through generating

**Comment by dakahn** [comment]
Ack! I think I've updated one too many files somewhere (should have stopped at 7/8 😆).

**Review by emyarod — COMMENTED** [review]
`danger` is being declared twice in the theme js files, causing CI to fail

**Comment by shixiedesign** [comment]
## Instruction for reviewers:
1. Confirm danger button component is using `$danger` token for background. 
2. All tests pass, no other components are affected.

**Comment by dakahn** [comment]
Added that to the PR description! Thanks @shixiedesign

**Comment by dakahn** [comment]
Requested some re-reviews since there were a handful of changes over the last few days. Thanks team 👍. And thanks for the assist with the error log debugging @emyarod 🙏

**Review by joshblack — APPROVED** [review]
Looks good, only question was around the generated stylesheet. Might help to let CI do that, or run `build` to generate the assets

**Comment by dakahn** [comment]
@joshblack _which generated stylesheet that I manually updated are you referring to_? :)

**Comment by joshblack** [comment]
@dakahn was wondering about this one: https://github.com/carbon-design-system/carbon/pull/4037/files#diff-8ad3b69a618b0df8bd35e59e4ae256ab

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @dakahn!

**Comment by joshblack** [comment]
@asudoh we were still discussing this comment: https://github.com/carbon-design-system/carbon/pull/4037#issuecomment-533696517

**Comment by asudoh** [comment]
Sorry for that, great to clarify that we should hold in such case.

**Pull request merged**